docs/features/software-templates/writing-templates.md

```diff
@@ -801,34 +801,33 @@ The `projectSlug` filter generates a project slug from a repository URL
 - **Input**: `github.com?repo=backstage&org=backstage`
 - **Output**: `backstage/backstage`
 
-## Custom Filters
+## Custom Filters and Globals
 
-Whenever it is needed to extend the built-in filters with yours `${{ parameters.name | my-filter1 | my-filter2 | etc }}`, then you can add them
-using the property `additionalTemplateFilters`.
+You may wish to extend the filters and globals with your own custom ones. For example `${{ myGlobal | myFilter | myOtherFilter }}` or `${{ myFunctionGlobal(1,2) | myFilter }}`.
+This can be achieved using the `additionalTemplateFilters` and `additionalTemplateGlobals` properties respectively.
 
-The `additionalTemplateFilters` property accepts as type a `Record`
+These properties accept a `Record`
 
-```ts title="plugins/scaffolder-backend/src/service/Router.ts"
+```ts title="plugins/scaffolder-backend/src/service/router.ts"
   additionalTemplateFilters?: Record<string, TemplateFilter>;
+  additionalTemplateGlobals?: Record<string, TemplateGlobal>;
 ```
 
-where the first parameter is the name of the filter and the second receives a list of `JSON value` arguments. The `templateFilter()` function must return a JsonValue which is either a Json array, object or primitive.
+where the first parameter is the identifier of the filter or global and the second is a `TemplateFilter` or a `TemplateGlobal` respectively.
+A `TemplateFilter` is a function which will be called using the previous `JsonValue` objets and may return a `JsonValue` object.
```

Comment:
Because json object and value are generic types, I found it hard to understand the format of the parameters and what this function should return. It would be nice to share a real-world example. For example: Imagine you have a use case X, so you can define a filter or global like this... This function receives a parameter Y, does Z with it, and returns A which can be used later in your template like this...

Reply:
I was orientating my changes with the existing docs. It might be a good idea to scrap the whole "this is how the template code looks like internally" bit and go with the usage instead.

Reply:
I like you idea, that would be awesome!